### PR TITLE
Refactor ts types

### PR DESCRIPTION
**Pull request by fflorent**
This PR updates type definitions, mostly based on file `src/internal/types.ts` of #1128.

### CODE REVIEW

File: src/Credentials.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
+import type { ICredentials } from './internal/type'
```
Comment (trim21) [inline src/Credentials.ts:1]: please add and always add real source file extension

File: src/helpers.ts [hunk 2]
```diff
@@ -17,8 +17,6 @@ import {
 import type { Encryption, ObjectMetaData, RequestHeaders } from './internal/type.ts'
 import { RETENTION_MODES } from './internal/type.ts'
 
-export { ENCRYPTION_TYPES, LEGAL_HOLD_STATUS, RETENTION_MODES, RETENTION_VALIDITY_UNITS } from './internal/type.ts'
```
Comment (trim21) [inline src/helpers.ts:20]: do not remove exports from `src/helpers.ts`, this may cause breaking change.

File: src/internal/xml-parser.ts [hunk 3]
```diff
@@ -5,7 +5,8 @@ import { XMLParser } from 'fast-xml-parser'
 import * as errors from '../errors.ts'
 import { parseXml, sanitizeETag, sanitizeObjectKey, toArray } from './helper.ts'
 import { readAsString } from './response.ts'
-import type { BucketItemFromList, BucketItemWithMetadata } from './type.ts'
+import type { BucketItemWithMetadata } from './type'
```
Comment (trim21) [inline src/internal/xml-parser.ts:8]: always add source file extension

File: src/Credentials.ts (hunk 1 above)
Comment (trim21) [reply src/Credentials.ts:1]: https://github.com/minio/minio-js/blob/master/CONTRIBUTING.md#style-guide

This is needed in the source so our build script can build valid package for all users, even esm and `"moduleResolution": "node16"`

Comment (fflorent) [reply src/Credentials.ts]: Sorry, that's my auto-completion that imported this type and omitted the file extension. I'll take a closer look next time to ensure the extension is always included.

That's fixed in a21c956

File: src/helpers.ts (hunk 2 above)
Comment (fflorent) [reply src/helpers.ts]: Fixed in 40be1c9

**Comment by aldy505** [comment]
Hi @trim21 @prakashsvmx @kaankabalak can we prioritize this first? Or is there any other stuff that's blocking this PR to be merged (other than code reviews)?

**Comment by fflorent** [comment]
Should I close this issue?

**Comment by prakashsvmx** [comment]
Yes. @fflorent . thank you so much.

**Pull request closed without merging**